### PR TITLE
Remove more Python 2 artifacts

### PR DESCRIPTION
**Pull request by cottsay**
We no longer support Python 2 in rosdistro so there's no reason to carry these snippets anymore.

### CODE REVIEW

File: scripts/rosdistro_build_cache [hunk 1]
```diff
@@ -94,11 +92,7 @@ def main():
         # write the cache
         data = yaml.dump(cache.get_data(), Dumper=CacheYamlDumper)
 
-        # On Python 3, we must encode the unicode yaml str prior to writing it,
-        # whereas for Python 2, the yaml output is a str which cannot be further
-        # encoded (compared with a unicode object).
-        if sys.version_info[0] >= 3:
-            data = data.encode('utf-8')
+        data = data.encode('utf-8')
```
Comment (mikepurvis) [inline scripts/rosdistro_build_cache:95]: data = yaml.dump(cache.get_data(), Dumper=CacheYamlDumper).encode()

Comment (cottsay) [reply scripts/rosdistro_build_cache:95]: Would omitting `'utf-8'` constitute a change in behavior on any platforms?

Comment (mikepurvis) [reply scripts/rosdistro_build_cache:95]: It's the default: https://docs.python.org/3/library/stdtypes.html#str.encode

Comment (mikepurvis) [reply scripts/rosdistro_build_cache:95]: A better way still might be to just let the encode happen implicitly by changing `open('wb')` to just `open('w')`, as it looks like gzip.open also has a textual mode that presumable does the same thing: https://docs.python.org/3/library/gzip.html#gzip.open

But I can also understanding wanting to keep this change simple. :)

Comment (cottsay) [reply scripts/rosdistro_build_cache]: I'll take the suggestion, but let's not rock the boat too much :) a1b4ba9

**Review by nuclearsandwich — APPROVED** [review]

**Pull request merged**